client/v2/autocli/flag/map_types.go

```diff
@@ -0,0 +1,63 @@
+package flag
+
+import (
+	"strings"
+
+	"github.com/cockroachdb/errors"
+)
+
+type genericMapValueOptions[K comparable, V any] struct {
+	keyParser   func(string) (K, error)
+	valueParser func(string) (V, error)
+	genericType string
+}
+
+type genericMapValue[K comparable, V any] struct {
+	value   *map[K]V
+	changed bool
+	Options genericMapValueOptions[K, V]
+}
+
+func newGenericMapValue[K comparable, V any](val map[K]V, p *map[K]V) *genericMapValue[K, V] {
+	ssv := new(genericMapValue[K, V])
+	ssv.value = p
+	*ssv.value = val
+	return ssv
+}
+
+func (gm *genericMapValue[K, V]) Set(val string) error {
+	ss := strings.Split(val, ",")
+	out := make(map[K]V, len(ss))
+	for _, pair := range ss {
+		kv := strings.SplitN(pair, "=", 2)
+		if len(kv) != 2 {
+			return errors.Errorf("%s must be formatted as key=value", pair)
+		}
+		var err error
```

Comment:
This is an unnecessary declaration, please delete it: on the line below we do `key, err := gm.Options...`